### PR TITLE
feat: support core regex flags

### PR DESCRIPTION
**Pull request by mdjastrzebski**
### Summary

Support core RegExp flags.

### CODE REVIEW

File: src/__tests__/builder.test.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  expect(buildRegex({ multiline: false }, 'a').flags).toBe('');
+
+  expect(buildRegex({ hasIndices: true }, 'a').flags).toBe('d');
+  expect(buildRegex({ hasIndices: false }, 'a').flags).toBe('');
+
+  expect(buildRegex({ sticky: true }, 'a').flags).toBe('y');
+  expect(buildRegex({ sticky: false }, 'a').flags).toBe('');
+
+  expect(
+    buildRegex({ global: true, ignoreCase: true, multiline: false }, 'a').flags
```
Comment (okwasniewski) [inline src/__tests__/builder.test.ts:23]: This looks great!

File: src/utils.ts [hunk 2]
```diff
@@ -33,3 +34,18 @@ export function concatNodes(nodes: EncoderNode[]): EncoderNode {
 export function escapeText(text: string) {
   return text.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
 }
+
+export function extractPropsParam<Props>(
```
Comment (okwasniewski) [inline src/utils.ts:38]: Small comment of what this is used for and why we need it would be great

**Review by okwasniewski — APPROVED** [review]

File: src/builders.ts [hunk 3]
```diff
@@ -22,3 +49,24 @@ export function buildPattern(
 ): string {
   return encodeSequence(elements).pattern;
 }
+
+function encodeFlags(flags: RegexFlags): string {
```
Comment (jaworek) [inline src/builders.ts:53]: Would it make sense to move `encodeFlags` to `encoder` folder to keep encoders together?

**Review by jaworek — APPROVED** [review]

File: src/utils.ts (hunk 2 above)
Comment (mdjastrzebski) [reply src/utils.ts]: I've removed that function altogether.

File: src/builders.ts (hunk 3 above)
Comment (mdjastrzebski) [reply src/builders.ts:53]: `encodeFlags` is only used in `buildRegex', so I would keep it there for now.

**Pull request merged**